src/components/ImageView/index.tsx

```diff
@@ -1,36 +1,62 @@
+import purify from 'dompurify';
 import log from 'loglevel';
 import * as React from 'react';
 
 import styles from './styles.module.scss';
 
-type PublicProps = {
+export type PublicProps = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   mimeType: string;
   content: string;
 };
 
 type ConvertImageDataParams = {
   _btoa?: typeof btoa;
   _log?: typeof log;
+  _sanitize?: typeof purify.sanitize;
   content: string;
+  mimeType: string;
 };
 
-export const convertImageData = ({
+const convertImageData = ({
   _btoa = btoa,
   _log = log,
+  _sanitize = purify.sanitize,
   content,
+  mimeType,
 }: ConvertImageDataParams): string | null => {
+  const mimeTypeCased = mimeType.toLowerCase();
+  if (!mimeTypeCased.startsWith('image')) {
+    throw new Error(`Unexpected mimeType: "${mimeType}"`);
+  }
+
   try {
-    return _btoa(content);
+    return _btoa(
+      // Run sanitize on svg files.
+      mimeTypeCased === 'image/svg+xml' ? _sanitize(content) : content,
+    );
   } catch (error) {
-    _log.debug('Error converting image data to ascii:', error);
+    _log.error('Error converting image data to ascii:', error);
```

Comment:
Thanks for the example patch. At first, I thought you were saying that there is an expected error that sometimes happens in real life.

Because we don't expect this to throw an error, I would like to see this try block removed so that the error gets thrown. This will show us the error screen you mentioned which has stack traces and code highlighting. This is what we want. If there is an error thrown by `btoa()` then that sounds like a bug we need to fix asap.